File: project_2/Main.java
class Main {
    public static void main(String[] args) {
        // Initialize the scanner with the input file
        Scanner S = new Scanner(args[0]);

        Prog prog = new Prog();

        if (S.currentToken() == Core.PROGRAM) {
            prog.parse(S);
            prog.print();
        }

    }
}

File: project_2/main.py
#!/usr/bin/env python3
from Interpreter import Interpreter
from Tokenizer import Tokenizer
from Core import Core
import sys

def main():
  # Initialize the tokenizer with the input file
  tokenizer = Tokenizer(sys.argv[1])
  interpreter = Interpreter(tokenizer)
  interpreter.parse()
  interpreter.validate()
  # interpreter.print()
  interpreter.execute()


if __name__ == "__main__":
    main()
